scripts/ci/ensure-angular-dependencies-compatibility.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * This script is in charge of making sure that the different modules
+ * that compose the AngularJS framework have compatible version.
+ *
+ * See: https://github.com/resin-io/etcher/pull/1168
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const EXIT_CODES = require('../../lib/shared/exit-codes');
+const packageJSON = require('../../package.json');
+
+const dependencies = [
+  {
+    name: 'angular',
+    value: packageJSON.dependencies.angular
+  },
+  {
+    name: 'angular-mocks',
+    value: packageJSON.devDependencies['angular-mocks']
+  }
+];
+
+_.each(dependencies, (dependency) => {
+  if (!/^\d+\.\d+\.\d+/.test(dependency.value)) {
+    console.error(`Dependency: ${dependency.name} should be locked`);
```

Comment:
Ahhh. Then (since these `sanity-checks` get run *after* `make electron-develop`) would it be worth checking that the installed `node_modules` match up pair-wise too?  (i.e. if we had `angular` 1.6.3 and `angular-mocks` 1.6.3 in `package.json`, but `angular` 1.6.5 and `angular-mocks` 1.6.5 installed into `node_modules`, presumably that'd work fine?)

Hmmm, and should `npm run sanity-checks;` actually get run **before** `npm test;`? Because being told "angular is version 1.6.1 but angular-mocks is version 1.6.3" is probably more useful error-output than "TypeError: angular.module(...).info is not a function" ? ;-)